lms/static/js/learner_dashboard/views/course_enroll_view_2017.js

```diff
@@ -50,33 +49,26 @@
                  },
 
                  handleEnroll: function() {
-                    // Enrollment click event handled here
-                     if (!this.model.get('course_run_key')) {
-                         this.$('.select-error').addClass('visible');
-                     } else if (!this.model.get('is_enrolled')) {
+                     // Enrollment click event handled here
+                     var courseRunKey = $('.run-select').val();
+                     this.model.updateCourseRun(courseRunKey);
+                     if (!this.model.get('is_enrolled')) {
                          // Create the enrollment.
                          this.enrollModel.save({
-                             course_id: this.model.get('course_run_key')
+                            course_id: courseRunKey
                          }, {
-                             success: _.bind(this.enrollSuccess, this),
-                             error: _.bind(this.enrollError, this)
+                            success: _.bind(this.enrollSuccess, this),
+                            error: _.bind(this.enrollError, this)
                          });
-                     }
-                 },
 
-                 handleCourseRunSelect: function(event) {
-                     var courseRunKey = $(event.target).val();
-
-                     if (courseRunKey) {
-                         this.model.updateCourseRun(courseRunKey);
-                     } else {
-                         // Set back the unselected states
-                         this.model.setUnselected();
                      }
                  },
 
                  enrollSuccess: function() {
                      var courseRunKey = this.model.get('course_run_key');
+                     window.analytics.track('Program Details Enrollment', {
```

Comment:
We use a specific format for events that we should stick to unless there's a good reason not to. Refer to the events in the edx.org GA account for an example.

Also: I don't think the `courseRun` property you're including here is going to be surfaced in GA. It may not be worth including.